src/components/HistoryTable/utils.ts

```diff
@@ -24,57 +33,83 @@ export const processCommits = (
       processedCommits.length > 0
         ? processedCommits[processedCommits.length - 1]
         : null;
-    if (commitType === rowType.COMMIT) {
-      const { version } = commit;
-      if (priorCommit && isSameDay(version.createTime, priorCommit.date)) {
-        processedCommits.push({
-          type: rowType.COMMIT,
-          commit: version,
-          date: version.createTime,
-          rowHeight: COMMIT_HEIGHT,
-        });
-      } else {
-        processedCommits.push({
-          type: rowType.DATE_SEPARATOR,
-          date: version.createTime,
-          rowHeight: DATE_SEPARATOR_HEIGHT,
-        });
-        processedCommits.push({
-          type: rowType.COMMIT,
-          commit: version,
-          date: version.createTime,
-          rowHeight: COMMIT_HEIGHT,
-        });
-      }
-    } else if (commitType === rowType.FOLDED_COMMITS) {
-      const { rolledUpVersions } = commit;
-      const firstRolledUpVersion = rolledUpVersions[0];
-      if (
-        priorCommit &&
-        isSameDay(firstRolledUpVersion.createTime, priorCommit.date)
-      ) {
-        processedCommits.push({
-          type: rowType.FOLDED_COMMITS,
-          rolledUpCommits: rolledUpVersions,
-          date: firstRolledUpVersion.createTime,
-          rowHeight: FOLDED_COMMITS_HEIGHT,
-        });
-      } else {
-        processedCommits.push({
-          type: rowType.DATE_SEPARATOR,
-          date: firstRolledUpVersion.createTime,
-          rowHeight: DATE_SEPARATOR_HEIGHT,
-        });
-        processedCommits.push({
-          type: rowType.FOLDED_COMMITS,
-          rolledUpCommits: rolledUpVersions,
-          date: firstRolledUpVersion.createTime,
-          rowHeight: FOLDED_COMMITS_HEIGHT,
-        });
+
+    switch (commitType) {
+      case rowType.COMMIT:
+        {
+          const { version } = commit;
+          if (priorCommit && isSameDay(version.createTime, priorCommit.date)) {
+            processedCommits.push({
+              type: rowType.COMMIT,
+              commit: version,
+              date: version.createTime,
+              selected: version.order === selectedCommitOrder,
+              rowHeight: COMMIT_HEIGHT,
+            });
+          } else {
+            processedCommits.push({
+              type: rowType.DATE_SEPARATOR,
+              date: version.createTime,
+              rowHeight: DATE_SEPARATOR_HEIGHT,
+            });
+            processedCommits.push({
+              type: rowType.COMMIT,
+              commit: version,
+              date: version.createTime,
+              selected: version.order === selectedCommitOrder,
+              rowHeight: COMMIT_HEIGHT,
+            });
+          }
+          if (version.order === selectedCommitOrder) {
+            selectedCommitRowIndex = processedCommits.length - 1;
+          }
+        }
```

Comment:
Bump on this: is there any reason we shouldn't consolidate `version.order === selectedCommitOrder` into a single variable, since it's repeated three times in this block?

Reply:
oops sorry missed this.